### PR TITLE
RDM-6167 Implement DSL part 1 review actions

### PR DESCRIPTION
**Pull request by bmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-6167

### Change description ###

- moved initialisation of test data to background step
- removed hooks class
- custom functional test exception
- added prefix to map verification for more accurate error reporting

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/fta/exception/FunctionalTestException.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.ccd.fta.exception;
+
+public class FunctionalTestException extends RuntimeException {
+
+    private final String validationMessage;
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/fta/exception/FunctionalTestException.java]: field redundant - super.getMessage() gives the same

File: src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java [hunk 2]
```diff
@@ -190,11 +202,11 @@ public void submitTheRequestToCallAnOperationOfAProduct(String operation, String
                 response = theRequest.delete(uri);
                 break;
             default:
-                Assert.fail("Unknown request method in data file");
+                throw new FunctionalTestException("Unknown request method in data file");
         }
 
         QueryableRequestSpecification queryableRequest = SpecificationQuerier.query(theRequest);
-        scenario.write(queryableRequest.getMethod() + " " + queryableRequest.getURI());
+        scenario.write("Calling " + queryableRequest.getMethod() + " on " + queryableRequest.getURI());
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java]: no " on " in between pls

File: src/aat/java/uk/gov/hmcts/ccd/fta/exception/FunctionalTestException.java (hunk 1 above)
Comment (bmcts) [reply src/aat/java/uk/gov/hmcts/ccd/fta/exception/FunctionalTestException.java]: 👍

File: src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java (hunk 2 above)
Comment (bmcts) [reply src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java]: 👍

**Pull request merged**